lib/xboxrt/libc_extensions/wchar.h

```diff
@@ -54,6 +57,20 @@ size_t wcslen( const wchar_t * s );
 int mbsinit( const mbstate_t * ps );
 size_t mbrtowc( wchar_t * XBOXRT_RESTRICT pwc, const char * XBOXRT_RESTRICT s, size_t n, mbstate_t * XBOXRT_RESTRICT ps );
 size_t wcrtomb( char * XBOXRT_RESTRICT s, wchar_t wc, mbstate_t * XBOXRT_RESTRICT ps );
+wchar_t *wmemset(wchar_t *wcs, wchar_t wc, size_t n);
+double wcstod(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr);
+float wcstof(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr);
+long double wcstold(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr);
+long wcstol(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr, int base);
+long long wcstoll(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr, int base);
+unsigned long wcstoul(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr, int base);
+unsigned long long wcstoull(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr, int base);
+size_t mbrlen(const char *s, size_t n, mbstate_t *ps);
+wint_t btowc(int c);
+size_t mbsrtowcs(wchar_t *dest, const char **src, size_t len, mbstate_t *ps);
+int wctob(wint_t c);
+int _snwprintf(wchar_t *buffer, size_t count, const wchar_t *format, ...);
```

Comment:
This looks like a Windows extension again. It should be separated from the rest, a comment should be added

Reply:
Non-C features should not only be separated, but also moved to an extension header / C file instead (wchar_ext_.h / wchar_ext_.c). I didn't even realize that upstream was missing wchar.h entirely.

We should then mimic the extension hook by including the wchar_ext_.h from our own wchar.h (which, hopefully, will eventually replaced by the pdclib wchar.h).

Reply:
I checked all declarations I added against the C standard, and as a result, moved `_snwprintf` into an extension header along with a comment.